caps/eth.md

```diff
@@ -108,22 +108,173 @@ relayed by the peer.
 Transactions must be validated before re-propagating them. Relaying an invalid transaction
 results in peer disconnection.
 
+### Transaction Encoding and Validity
+
+Transaction objects exchanged by peers have one of two encodings, which are identical to
+the encodings used by Ethereum consensus. In definitions across this specification, we
+refer to transaction objects of either encoding using the identifier `txₙ`.
+
+    tx = {legacy-tx, typed-tx}
+
+Untyped, legacy transactions are given as an RLP list.
+
+    legacy-tx = [
+        nonce: P,
+        gas-price: P,
+        gas-limit: P,
+        recipient: {B_0, B_20},
+        value: P,
+        data: B,
+        V: P,
+        R: P,
+        S: P,
+    ]
+
+[EIP-2718] typed transactions are encoded as RLP byte arrays where the first byte is the
+transaction type (`tx-type`) and the remaining bytes are opaque type-specific data.
+
+    typed-tx = rlp(tx-type || tx-data)
+
+Transactions must be validated when they are received. Validity depends on the Ethereum
+chain state. While the encoding of typed transactions is opaque, it is assumed that the
+encoding of `tx-data` provides values for `nonce`, `gas-price`, `gas-limit`, and that the
+sender account of the transaction can be determined from the signature.
+
+Transactions must be validated according to these rules:
+
+- If the transaction is typed, the `tx-type` must be one of the types considered
+  acceptable at the current block height.
```

Comment:
I believe we decided this rule was "bad" to have.  Instead, the rule should be "if the transaction type is not known by the client, it should be rejected with prejudice".  The problem is that near a fork block, we may start seeing transactions of the new type show up sooner than we see the fork block show up just due to block propagation delays.  In such a scenario, we don't want to kick peers because they happened to see a fork block before us.

By instead just requiring that the transaction type be *known* to the client, it allows us to transmit/propagate typed transactions leading up to and just before a fork block.  For example, if you get a typed transaction immediately before a fork block (due to just not having seen the fork block yet), and your client doesn't know how to handle it then you are going to get kicked off the main chain in a few seconds anyway so de-peering you is about to happen.